apps/backend/src/app/modules/users/users.service.spec.ts

```diff
@@ -0,0 +1,286 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { UsersService } from './users.service';
+import { mockDeep } from 'jest-mock-extended';
+import {
+  PRISMA_MOCK_PROVIDER,
+  PrismaMock
+} from '../../../../test/prisma-mock.const';
+import { EXTENDED_PRISMA_SERVICE } from '../database/db.constants';
+import { BadRequestException } from '@nestjs/common/exceptions/bad-request.exception';
+import {
+  ForbiddenException,
+  InternalServerErrorException,
+  ServiceUnavailableException
+} from '@nestjs/common';
+import { User } from '@prisma/client';
+import { AuthenticatedUser } from '../auth/auth.interface';
+import { SteamUserSummaryData } from '../steam/steam.interface';
+import { UsersGetAllQueryDto } from '../../dto';
+
+describe('UserService', () => {
+  let usersService: UsersService;
+  let db: PrismaMock;
+
+  beforeAll(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      providers: [UsersService, PRISMA_MOCK_PROVIDER]
+    })
+      .useMocker(mockDeep)
+      .compile();
+    usersService = module.get(UsersService);
+    db = module.get(EXTENDED_PRISMA_SERVICE);
+  });
+
+  it('should be defined', () => {
+    expect(usersService).toBeDefined();
+    expect(db).toBeDefined();
+  });
+
+  describe('getAll', () => {
+    it('should throw an error when passing both steamID and steamIDs', async () => {
+      const a: UsersGetAllQueryDto = {
+        steamID: '123456789',
+        steamIDs: ['123456789', '999999999']
+      };
+      await expect(usersService.getAll(a)).rejects.toThrow(BadRequestException);
+    });
+  });
+
+  describe('findOrCreateFromGame', () => {
+    it("should throw an error when the steamid's don't match", async () => {
+      const steamUserSummaryData: SteamUserSummaryData = {
+        avatar: '',
+        avatarfull: '',
+        avatarhash: '',
+        avatarmedium: '',
+        communityvisibilitystate: 0,
+        lastlogoff: 0,
+        loccountrycode: '',
+        personaname: '',
+        personastate: 0,
+        personastateflags: 0,
+        primaryclanid: '',
+        profilestate: 0,
+        profileurl: '',
+        realname: '',
+        timecreated: 0,
+        steamid: '123456789'
+      };
+      jest
+        .spyOn(usersService['steamService'], 'getSteamUserSummaryData')
+        .mockImplementationOnce(() => Promise.resolve(steamUserSummaryData));
+      await expect(
+        usersService.findOrCreateFromGame(BigInt(999999999))
+      ).rejects.toThrow(BadRequestException);
+    });
+    it('should throw an error when getSteamUserSummaryData throws and error', async () => {
```

Comment:
Nit: these files can get dense as it is, really worth adding an extra newline between outermost function calls and similar like this.